datastore/memory/txn.go

```diff
@@ -123,9 +123,9 @@ func (t *basicTxn) GetSize(ctx context.Context, key ds.Key) (size int, err error
 
 // Has implements ds.Has.
 func (t *basicTxn) Has(ctx context.Context, key ds.Key) (exists bool, err error) {
-	t.ds.closeLk.RLock()
-	defer t.ds.closeLk.RUnlock()
-	if t.ds.closed {
+	t.closeLk.RLock()
```

Comment:
For reference, for myself and other reviewers, the rest of the funcs in `basicTxn` do this (`t.closeLk`) and this appears to be a bug fix.